File: fn/src/main/java/org/sqlbuilder/fn/RequiresIdFrom.java
package org.sqlbuilder.fn;

import java.lang.annotation.ElementType;
import java.lang.annotation.Repeatable;
import java.lang.annotation.Target;

@Repeatable(RequiresIdsFrom.class)
@Target({ElementType.METHOD, ElementType.CONSTRUCTOR })
public @interface RequiresIdFrom
{
	Class<?> type();
}

File: reqprov/src/main/java/org/sqlbuilder/annotations/RequiresIdFrom.kt
package org.sqlbuilder.annotations

import kotlin.reflect.KClass

@JvmRepeatable(RequiresIdsFrom::class)
@Target(
    AnnotationTarget.FUNCTION,
    AnnotationTarget.PROPERTY_GETTER,
    AnnotationTarget.PROPERTY_SETTER,
    AnnotationTarget.CONSTRUCTOR
)
annotation class RequiresIdFrom
    (val type: KClass<*>)
